bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/dto/clip2/ButtonReport.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.hue.internal.dto.clip2;
+
+import java.time.Instant;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.hue.internal.dto.clip2.enums.ButtonEventType;
+
+/**
+ * DTO for CLIP 2 button report.
+ *
+ * @author Jacob Laursen - Initial contribution
+ */
+@NonNullByDefault
+public class ButtonReport {
```

Comment:
@jlaur can you please inherit all the `xyzReport` classes from a common `BaseReport` class as follows? Reason is that there is no need for the `changed` fields and `getLastChanged()` methods to be replicated so many times across all the inherited `xyzReport` class instances. Also I rely on that same `BaseReport` class for the home security additions.

```
@NonNullByDefault
public abstract class BaseReport {

    private @NonNullByDefault({}) @SerializedName(value = "changed", alternate = { "updated" }) Instant changed;

    public Instant getLastChanged() {
        return changed;
    }
}
```

Reply:
Please see https://github.com/openhab/openhab-addons/pull/15552#discussion_r1328076737 where I already commented this proposal.